File: Cc.Ui/src/models/tabRegion.ts
import { Entity, resource, field, association } from '../framework/entity';
import { Pagination, Pager } from '../framework/core/actions';

@resource('tabRegion')
export class TabRegion extends Entity {

    @field()
    id: number;
    @field()
    insertDate: string;
    @field()
    insertUser: string;
    @field()
    updateDate: string;
    @field()
    updateUser: string;
    
    @field()
    code: string;
    @field()
    description: string;
    @field()
    nationCode: string;
    
    guid: string;

    constructor(data?: any) {
        super()
        this.init(data);      
    }

    static newItem(): TabRegion {

        let item = new TabRegion();
        return item;
    }
}

@resource('tabRegion')
export class TabRegionSearchModel extends Entity implements Pagination {
        
    @field()
    pager: Pager = <Pager> {
        take: 100,
        orderBy: 'id'
    };
}

File: Cc.Core/Models/TabRegion.cs
using System;
using System.Collections.Generic;

namespace Cc.Core.Models;

public partial class TabRegion
{
    public long Id { get; set; }

    public DateTime InsertDate { get; set; }

    public string? InsertUser { get; set; }

    public DateTime UpdateDate { get; set; }

    public string? UpdateUser { get; set; }

    public string? Code { get; set; }

    public string? Description { get; set; }

    public string NationCode { get; set; } = null!;

    public virtual ICollection<BroadcastingTabRegion> BroadcastingTabRegion { get; set; } = new List<BroadcastingTabRegion>();

    public virtual ICollection<PrincipalTabRegion> PrincipalTabRegion { get; set; } = new List<PrincipalTabRegion>();

    public virtual ICollection<TabDistrict> TabDistrict { get; set; } = new List<TabDistrict>();
}
